### PR TITLE
feat: implemented/added scroll-area component to design-system 

### LINKED ISSUE #1273: bug: Python is not in the list
I can't upload an image but the filters on the dashboard does include Python

### PR DESCRIPTION
**Pull request by babblebey**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR Implements `@radix-ui/react-scroll-area` primitive to create the `ScrollArea` component which was in turn implemented in the `Select` component to allow scrolling of overflowing `SelectItems` value within the component which wasn't caught by the `avoidCollisions` prop because of our preferred `position` value i.e. `popper`

I had initially intended to address this with the [ScrollUpButton/ScrollDownButton](https://www.radix-ui.com/docs/primitives/components/select#scrollupbutton) sub-component of the already-implemented `@radix-ui/react-select` primitive, but there was a certain challenge with this approach.

In order for to use the `ScrollUpButton` and `ScrollDownButton` components, we'd need to have the `position` prop of the Select Primitive Content set to the default `item-aligned` instead of our preferred `popper` (as seen below)

Code Sample
```javascript
// components\atoms\Select\select.tsx

<SelectPrimitive.Content
    ref={ref}
    ....
    // 'popper' set as default via component props
    position="popper"
    ...
  >
  <SelectPrimitive.Viewport>
    {children}
  </SelectPrimitive.Viewport>
</SelectPrimitive.Content>
```

But doing this, i.e. setting the Select Primitive Content `position` to the default `item-aligned` it makes the content behave similarly to a native MacOS menu by positioning it relative to the active item as seen below on a responsive layout.

Unlike our current preferred `position` of `popper` which positions the content below the select trigger making it behave like a "dropdown menu" or "popover" as seen below on a responsive layout.

At the point when `position` defaults/sets to `item-aligned`, the `ScrollUpButton` and `ScrollDownButton` components, if implemented displays on demand when the content collides with the top/bottom of the viewport, as seen below on a responsive layout.

screencast-localhost_3000-2023 06 21-23_37_35

But this come at the expense of our preferred `position`ing value, enforcing the `item-aligned` which doesn't really seem to look aesthetically good nor great UX-wise. 

**The Trade-off**

I thought it was a better to stick with our preferred `position` value of `popper`, coupling it with the `ScrollArea` component implementation, we let our `Select` component behave like a dropdown menu, and also improve the usability experience by enhancing its general responsiveness, as seen in screenshot below.

screencast-localhost_3000-2023 06 22-14_45_23 (1)

## Related Tickets & Documents

fixes #1273 

## Mobile & Desktop Screenshots/Recordings

**Before**
[screencast-localhost_3000-2023.06.21-17_53_10.webm](https://github.com/open-sauced/insights/assets/25631971/915b9e91-d065-4780-a38a-a9c2914bb07b)

**After**
[screencast-localhost_3000-2023.06.21-17_38_47.webm](https://github.com/open-sauced/insights/assets/25631971/cc44a05d-53b1-4972-9bef-8a33accfd798)

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
Nothing much, Its important to note that I added a new package though i.e. `@radix-ui/react-scroll-area`

### CODE REVIEW

File: components/atoms/ScrollArea/scroll-area.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ScrollViewport.displayName = ScrollAreaPrimitive.Viewport.displayName;
+
+const ScrollBar = React.forwardRef<
+  React.ElementRef<typeof ScrollAreaPrimitive.Scrollbar>,
+  React.ComponentPropsWithoutRef<typeof ScrollAreaPrimitive.Scrollbar>
+>(({ className, orientation = "vertical", ...props }, ref) => (
+  <ScrollAreaPrimitive.Scrollbar
+    ref={ref}
+    className={clsx(
+      "flex touch-none select-none p-[1px] transition-colors duration-[160ms] ease-out",
```
Comment (NsdHSO) [inline components/atoms/ScrollArea/scroll-area.tsx:53]:
```suggestion
      "flex touch-none select-none p-0.5 transition-colors duration-[160ms] ease-out",
```

.p-0.5 => 2px

File: stories/atoms/scroll-area.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+Default.args = {
+  type: "auto",
+  style: {
+    height: "200px",
+    width: "350px"
+  }
+}
+
+const SelectOptions = Array.from({ length: 50 }).map((_, i, a) => `v1.2.0-beta.${a.length - i}`)
```
Comment (NsdHSO) [inline stories/atoms/scroll-area.stories.tsx:42]:
```suggestion
const SelectOptions = Array.from({ length: 50 }).map((value, index, array) => `v1.2.0-beta.${array.length - index}`);


```

File: components/atoms/ScrollArea/scroll-area.tsx (hunk 1 above)
Comment (babblebey) [reply components/atoms/ScrollArea/scroll-area.tsx:53]: Well noted

**Review by OgDev-01 — APPROVED** [review]
LGTM 🍕🍕

**Review by NsdHSO — APPROVED** [review]
LGTM

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**